Example/Auth/Sample/MainViewController.m

```diff
@@ -2458,35 +2521,49 @@ - (void)signInWithPhoneNumber {
   }];
 }
 
-/** @fn signInWithPhoneNumberRecaptcha
-    @brief Allows sign in with phone number using reCAPTCHA
+/** @fn signInWithPhoneNumber
+    @brief Allows sign in with phone number using reCAPTCHA.
+    @param phoneNumber Number pass in string.
+    @completion A completion block to be executed after successful phone number sign in.
  */
-- (void)signInWithPhoneNumberRecaptcha {
-  [self commonPhoneNumberInputWithTitle:@"Phone #" Completion:^(NSString *_Nullable phone) {
-    [self showSpinner:^{
-      [[AppManager phoneAuthProvider] verifyPhoneNumber:phone
-                                             UIDelegate:nil
-                                             completion:^(NSString *_Nullable verificationID,
-                                                          NSError *_Nullable error) {
-        [self hideSpinner:^{
-          if (error) {
-            [self logFailure:@"failed to send verification code" error:error];
-            [self showMessagePrompt:error.localizedDescription];
-            return;
+- (void)signInWithPhoneNumber:(NSString *_Nullable)phoneNumber
+                   completion:(void(^)(NSError *_Nullable))completion {
+  [self showSpinner:^{
+    [[AppManager phoneAuthProvider] verifyPhoneNumber:phoneNumber
+                                           UIDelegate:nil
+                                           completion:^(NSString *_Nullable verificationID,
+                                                        NSError *_Nullable error) {
+      [self hideSpinner:^{
+        if (error) {
+          [self logFailure:@"failed to send verification code" error:error];
+          [self showMessagePrompt:error.localizedDescription];
+          completion(error);
+          return;
+        }
+        [self logSuccess:@"Code sent"];
+        [self commonPhoneNumberInputWithTitle:@"Code"
+                                   Completion:^(NSString *_Nullable verificationCode) {
+         [self commontPhoneVerificationWithVerificationID:verificationID
+                                         verificationCode:verificationCode];
+          if (completion) {
+            completion(nil);
           }
-          [self logSuccess:@"Code sent"];
-
-          [self commonPhoneNumberInputWithTitle:@"Code"
-                                     Completion:^(NSString *_Nullable verificationCode) {
-            [self commontPhoneVerificationWithVerificationID:verificationID
-                                            verificationCode:verificationCode];
-          }];
         }];
       }];
     }];
   }];
 }
 
+/** @fn signInWithPhoneNumberPrompt
+    @brief Allows sign in with phone number via popup prompt.
+ */
+- (void)signInWithPhoneNumberPrompt {
```

Comment:
signInWithPhoneNumberWithPrompt sounds better.  The current name makes the method sound like a it only shows a prompt for signing in with phone number but not actually doing the sign in.  Same for other methods.

Reply:
done.